pkg/traversal/traversal.go

```diff
@@ -0,0 +1,133 @@
+package traversal
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/openshift/must-gather-clean/pkg/obfuscator"
+	"github.com/openshift/must-gather-clean/pkg/omitter"
+	"github.com/openshift/must-gather-clean/pkg/output"
+)
+
+type FileWalker struct {
+	input       string
+	obfuscators []obfuscator.Obfuscator
+	writer      output.Outputter
+	omitters    []omitter.Omitter
+}
+
+// NewFileWalker returns a FileWalker. It ensures that the input directory exists and is readable.
+func NewFileWalker(inputDir string, writer output.Outputter, obfuscators []obfuscator.Obfuscator, omitters []omitter.Omitter) (*FileWalker, error) {
+	fileInfo, err := os.Stat(inputDir)
+	if err != nil {
+		return nil, fmt.Errorf("cannot access input directory %s: %w", inputDir, err)
+	}
+	if !fileInfo.IsDir() {
+		return nil, fmt.Errorf("input path %s is not a directory", inputDir)
+	}
+	inputPath, err := filepath.Abs(inputDir)
+	if err != nil {
+		return nil, err
+	}
+	return &FileWalker{input: inputPath, obfuscators: obfuscators, writer: writer, omitters: omitters}, nil
+}
+
+// Traverse should be called to start processing the input directory.
+func (w *FileWalker) Traverse() error {
+	return w.processDir(w.input, "")
+}
+
+func (w *FileWalker) processDir(inputDirName string, outputDirName string) error {
+	// list all the entities in the directory
+	entries, err := os.ReadDir(inputDirName)
+	if err != nil {
+		return err
+	}
+	for _, e := range entries {
+		fileInfo, err := e.Info()
+		if err != nil {
+			return err
+		}
+
+		// If a file is a symlink then ignore it
+		if fileInfo.Mode()&os.ModeSymlink != 0 {
```

Comment:
good idea, can we add a test for that?

Reply:
Already exists.

Reply:
where? I don't see a symbolic link in the change list

Reply:
[Here](https://github.com/openshift/must-gather-clean/blob/80e178188f7e47b6220a256e6078f220f5f836fb/pkg/traversal/testfiles/test1/mg/nodes/another.yaml)

Reply:
:+1: the diff didn't show it's a symlink, thx!